src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -255,9 +266,8 @@ public void mouseMoved(EditorMouseEvent e) {
             return;
         }
         Language language = psiFile.getLanguage();
-        if ((!LanguageDocumentation.INSTANCE.allForLanguage(language).isEmpty() && !language
-                .equals(PlainTextLanguage.INSTANCE)) || (!getIsCtrlDown() && !EditorSettingsExternalizable
-                .getInstance().isShowQuickDocOnMouseOverElement())) {
+        if ((!LanguageDocumentation.INSTANCE.allForLanguage(language).isEmpty() && !isPlainTextLanguage(language))
```

Comment:
Could you please explain the reason behind this cache based new check? 
`language.equals(PlainTextLanguage.INSTANCE)` isn't working for you?

Reply:
Languages like java doesn't inherit thr plain text language. Thats why this new check is added.

Reply:
Is it not working only for java? how about using a `isFileSupported()` check from our `FileUtils` instead, because you are registering the command server denition for `java` file extensions right ? :)

Reply:
@NipunaRanasinghe i think this is a really good suggestion. I have changed the code now to use FileUtils check.

Reply:
@gayanper great :)